rust/lance/src/dataset/updater.rs

```diff
@@ -31,6 +31,11 @@ use crate::dataset::FileFragment;
 /// To use, call [`Updater::next`] to get the next [`RecordBatch`] as input,
 /// then call [`Updater::update`] to update the batch. Repeat until
 /// [`Updater::next`] returns `None`.
+///
+/// `write_schema` dictates the schema of the new file, while `final_schema` is
+/// the schema of the full fragment after the update. These are optional and if
+/// not specified, the updater will infer the write schema from the first batch
+/// of results and will append them to the current schema to get the final schema.
```

Comment:
Is the updater part of the public API I guess?  Or are there times that we won't specify the schema and times that we will?

Reply:
It is part of the public API, though it's a low-level API.

It was originally written to always infer the schema, and that's used in the `Fragment.add_columns()` API.